tests/Constraints/CoerciveTest.php

```diff
@@ -11,271 +11,201 @@
+                array('string',     '"true"',       true,           true),  // #15
+                array('integer',    '1',            true,           true),  // #16
+                array('boolean',    'true',         true,           true),  // #17
+                array('NULL',       'null',         false,          true),  // #18
+                array('array',      '["true"]',     true,           true),  // #19
+                array('object',     '{"a":"b"}',     null,          false), // #20
+                array('string',     '""',           null,           false), // #21
+                array('string',     '"ABC"',        null,           false), // #22
+                array('integer',    '2',            null,           false), // #23
+            ),
+            'NULL' => array(
+                array('string',     '""',           null,           true),  // #24
+                array('integer',    '0',            null,           true),  // #25
+                array('boolean',    'false',        null,           true),  // #26
+                array('NULL',       'null',         null,           true),  // #27
+                array('array',      '[0]',          null,           true),  // #28
+                array('object',     '{"a":"b"}',    null,           false), // #29
+                array('string',     '"null"',       null,           false), // #30
+                array('integer',    '-1',           null,           false), // #31
+            ),
+            'array' => array(
+                array('string',     '"ABC"',        array('ABC'),   true),  // #32
+                array('integer',    '45',           array(45),      true),  // #33
+                array('boolean',    'true',         array(true),    true),  // #34
+                array('NULL',       'null',         array(null),    true),  // #35
+                array('array',      '["ABC"]',      array('ABC'),   true),  // #36
+                array('object',     '{"a":"b"}',    null,           false), // #37
+            ),
+        );
 
-        $validator->validate($value, $schema, $checkMode);
+        $tests = array();
+        foreach ($types as $toType => $testCases) {
+            foreach ($testCases as $testCase) {
+                $tests[] = array(
+                    sprintf('{"properties":{"propertyOne":{"type":"%s"}}}', strtolower($toType)),
+                    sprintf('{"propertyOne":%s}', $testCase[1]),
+                    $testCase[0],
+                    $toType,
+                    $testCase[2],
+                    $testCase[3]
+                );
+            }
+        }
 
-        $this->assertTrue(gettype($value->number) == 'double');
-        $this->assertTrue(gettype($value->integer) == 'integer');
-        $this->assertTrue(gettype($value->negativeInteger) == 'integer');
-        $this->assertTrue(gettype($value->boolean) == 'boolean');
+        // #38 check post-coercion validation (to array)
+        $tests[] = array(
```

Comment:
They can. The reason I did it this way is because complex tests are more likely to be added than simple ones, and adding things to the beginning or midway through requires manually renumbering all the test cases - whereas adding something to the end is easy, and doesn't require renumbering anything.